lib/recipes-data/src/lib/extract-recipes.ts

```diff
@@ -8,30 +9,59 @@ import type {RecipeReferenceWithoutChecksum} from './models';
 export function extractAllRecipesFromArticle(content: Content): RecipeReferenceWithoutChecksum[] {
   if (content.type == ContentType.ARTICLE && content.blocks) {
     const articleBlocks: Blocks = content.blocks
-    const getAllMainBlockRecipesIfPresent: RecipeReferenceWithoutChecksum[] = extractRecipeData(content.id, articleBlocks.main as Block)
-    const bodyBlocks = articleBlocks.body as Block[]
-    const getAllBodyBlocksRecipesIfPresent: RecipeReferenceWithoutChecksum[] = bodyBlocks.flatMap(bodyBlock => extractRecipeData(content.id, bodyBlock))
+    const getAllMainBlockRecipesIfPresent: RecipeReferenceWithoutChecksum[] = extractRecipeData(content.id, articleBlocks.main)
+    const bodyBlocks = articleBlocks.body
+    const getAllBodyBlocksRecipesIfPresent: RecipeReferenceWithoutChecksum[] = bodyBlocks? bodyBlocks.flatMap(bodyBlock => extractRecipeData(content.id, bodyBlock)) : [];
     return getAllMainBlockRecipesIfPresent.concat(getAllBodyBlocksRecipesIfPresent)
   } else {
     return Array<RecipeReferenceWithoutChecksum>()
   }
 }
 
-export function extractRecipeData(canonicalId: string, block: Block): RecipeReferenceWithoutChecksum[] {
+export function extractRecipeData(canonicalId: string, block?: Block): RecipeReferenceWithoutChecksum[] {
+  if(! block?.elements) return [];
+
   return block.elements
     .filter(elem => elem.type === ElementType.RECIPE)
     .map(recp => parseJsonBlob(canonicalId, recp.recipeTypeData?.recipeJson as string))
     .filter(recp => !!recp) as RecipeReferenceWithoutChecksum[]
 }
 
-function parseJsonBlob(canonicalId: string, recipeJson: string): RecipeReferenceWithoutChecksum | null {
-  const recipeData = JSON.parse(recipeJson) as Record<string, unknown>
-  if (!recipeData.id) {
-    return null //TODO: we should incorporate a metric for failed recipes so we can have an indication of upstream issues.
+/**
+ * Most recipes have a UUID-style `id` field, so we pass that through.
+ * However some of the ones that were extracted by D&I have a numeric field still.
+ * In that case, we concatenate the canonical ID onto the numeric value and then sha1 the lot.
+ * @param contentIdField
```

Comment:
Might be wise to rename this as it's not the content ID as we typically refer to it - it's the recipe ID it seems